pkg/asset/installconfig/aws/validation.go

```diff
@@ -255,6 +255,34 @@ func validateMachinePool(ctx context.Context, meta *Metadata, fldPath *field.Pat
 			allErrs = append(allErrs, field.Invalid(fldPath.Child("type"), pool.InstanceType, errMsg))
 		}
 	}
+
+	allErrs = append(allErrs, validateSecurityGroupIDs(ctx, meta, fldPath.Child("additionalSecurityGroupIDs"), platform, pool)...)
+
+	return allErrs
+}
+
+func validateSecurityGroupIDs(ctx context.Context, meta *Metadata, fldPath *field.Path, platform *awstypes.Platform, pool *awstypes.MachinePool) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(pool.AdditionalSecurityGroupIDs) > 0 {
```

Comment:
Nit: If we invert the check we can reduce the indent level by 1
```suggestion
    // Nothing to validate
	if len(pool.AdditionalSecurityGroupIDs) == 0 {
	    return allErrs
	}
```